jsonschema-core.xml

```diff
@@ -1387,8 +1385,32 @@
                         without requiring JSON Pointer references to be updated.
                     </t>
                     <t>
-                        The "$anchor" keyword is used to specify such a fragment.  It is an
-                        identifier keyword that can only be used to create plain name fragments.
+                        The "$anchor" and "$dynamicAnchor" keywords are used to specify such
+                        fragments.  They are identifier keywords that can only be used to create
+                        plain name fragments, rather than absolute URIs as seen with "$id".
+                        The behavior of the created fragment is identical for both keywords.
+                    </t>
+                    <t>
+                        The base URI to which the resulting fragment is appended is the canonical
+                        URI of the schema resource containing the "$anchor" or "$dynamicAnchor"
+                        in question.  As discussed in the previous section, this is either the
+                        nearest "$id" in the same or parent schema object, or the base URI
+                        for the document as determined according to RFC 3986.
+                    </t>
+                    <t>
+                        Separately from the usual usage of URIs, "$dynamicAnchor"
+                        indicates that the fragment is an extension point when used with
+                        the "$dynamicRef" keyword.  This low-level, advanced feature
+                        makes it easier to extend recursive schemas such as the meta-schemas,
+                        without imposing any particular semantics on that extension.
+                        See the section on <xref target="dynamic-ref">"$dynamicRef"</xref>
+                        for details.
+                    </t>
+                    <t>
+                        In most cases, the normal fragment behavior both sufficies and
+                        is more intuitive.  Therefore it is RECOMMENDED that "$anchor"
```

Comment:
"more readable"? "less confusing"? I'm not sure that $recursiveAnchor is less *intuitive*, as much as more confusing and causes more cognitive (and parsing) overhead.

Reply:
I think this is fine. We can bikeshed phrasing later if required.